Fixes to Zed lake design doc after following as a user (#3014)
brimdata/zed: docs/lake/design.md

Data can be efficiently accessed via range scans composed of a
range of values conforming to the pool key.

> A pool key has a configured sort order, either ascending or descending

Author: @nwt: Ok, but if that's how it's to be seen, the help text perhaps could use some improvement.

```
$ zed lake create -h
...
OPTIONS
    -orderby comma-separated pool keys with optional :asc or :desc suffix to organize data in pool (cannot be changed) (default "ts:desc")
```

To me that reads as if the "pool key" is the noun attached to `-orderby` and the suffix is further describing said pool key. But I'm happy to merge your suggested wording in the design doc.

* a _commit_ operation is like a `git commit`,
* and a pool _snapshot_ is like a `git checkout`.

A core theme of the Zed lake design is _ergonomics_.  Given the Git metaphor,

> our goal here is that the Zed lake tooling be as easy and familiar as Git is

Author: I've seen @nwt propose corrections to capitalize "Git" when speaking of the tool, which I support, so I'm continuing in that spirit. I also support `git` being in lowercase & fixed width when we're referencing the command that's invoked at the shell, and it seems the doc is already quite consistent on that one.

```
To understand the log contents, the `load` operation is actually
decomposed into two steps under the covers:
an "add" step stores one or more

> new immutable data objects in the lake and a "commit" operation

Author: I'm normally a fan of invoking the fixed width font in backticks to add emphasis. However, since we used to have formal `zed lake` commands called `add` and `commit`, but they're gone, I figure _not_ putting them in fixed with helps contrast with the `load` command we just mentioned that _does_ remain.

Reviewer: Correct, this is described steps in the algorithm and isn't a command.

> new immutable data objects in the lake and a "commit" operation

Reviewer 2: Feels odd to say there are two steps but then call one an operation.
```suggestion
new immutable data objects in the lake and a "commit" step
```

Data is read from one or more pools with the `query` command.  The pool/branch names
are specified with `from` at the beginning of the Zed query along with an optional

> time range using `range` and `to`.  The default output format is ZSON (if to a
> terminal) or ZNG (if redirected elsewhere) though this can be overridden with

Reviewer 2: ```suggestion
time range using `range` and `to`.  The default output format is ZSON for
terminals and ZNG otherwise, though this can be overridden with
```

optimize layout for performance.

These structures are introspected using meta-queries that simply

> specify a metadata source using an extended syntax in the `from` operator.

Author: I remember seeing the "bracket syntax" in some earlier revisions, but it looks like it's gone in favor of the colon/at syntax. My choice of the word "extended" was kind of arbitrary, so please propose a correction if you'd prefer something else!

navigated by time.  For example, a list of branches of a pool's past
can be created by scanning the branches log and stopping at the largest
timestamp less than or equal to the desired timestamp.  Likewise, a branch

> can be located in a similar fashion, then its corresponding commit object

Author: I'm not sure if the comma is strictly necessary, but before it was there I paused on this sentence too long thinking "then" should be "than".

### Delete

Data objects can be deleted with the `delete` command.  This command
simply removes the data from the branch without actually deleting the
underlying data objects thereby allowing time travel to work in the face
of deletes.

> For example, this command deletes the three objects referenced

Author: We discussed this change on Slack. The help text still speaks as if `delete` would accept commits, so I mentioned that in #3012 along with the request to add a test for it.

> whatever reason, it should simply continue on and issue a warning on
> the query endpoint "warnings channel".

### Revert

The actions in a commit can be reversed with the `revert` command.  This
command applies the inverse steps in a new commit to the tip of the indicated
branch.  Any data loaded in a reverted commit remains in the lake but no longer

> appears in the branch.  The new commit may recursively be reverted by an

Reviewer 2: ```suggestion
appears in the branch.  The new commit may itself be reverted by an
```

additional revert operation.

For example, this command reverts the commit referenced by commit ID
`<commit>`.
```
zed lake revert -p logs <commit>
```

> 

Author: At the same time I was educated on Slack about `delete` only applying to objects, I was reminded of the existence of `revert`, at which point I noticed it's absent from the doc. I just copied this text pretty much verbatim from the `zed` help text. Feel free to propose more adjustments to suit the narrative nature of this doc.

```
zed lake index apply -p logs IndexGroupEx <tag>
```

> The index is created and a transaction put (somewhere).  Once this transaction

Author: This one needs genuine input! I just put some garbage text in there to trigger the PR suggestion workflow.

I know that "staging" as a formal construct is gone, but I don't know what took its place in this context. I know indexing in general is not operational (#2532), but I figure if we're going to make forward-looking statements about how it's expected to work, it might as well be accurate. :smile:

Reviewer: The index is created and transactionally committed to the branch...

* query: `from commitish:meta` branch or commit meta-scan (no defaulting to main here as that would be pool-level meta)
* query: `from pool:meta`
* query: `from :meta`
* rebase `ref` `commitish` - rebase a branch `ref` onto the commit object history starting at `commitish` (this is peculiar because `ref` could imply the pool name for `commitish` though this will usually be a branch name)
* rename `pool` `pool` - change name of a data pool

> * revert `ref` `commitish`- undo the commit at `commitish` with a new commit object and updates the branch `ref` to point at the new commit; typically this commit would be in the branches history but it doesn't have to be (but would typically fail a consistency check if it isn't in the history).  We could prevent this condition with a check.

Reviewer 2: ```suggestion
* revert `ref` `commitish`- undo the commit at `commitish` with a new commit object and update the branch `ref` to point at the new commit; typically this commit would be in the branch's history but it doesn't have to be (but would typically fail a consistency check if it isn't in the history).  We could prevent this condition with a check.
```
